### PR TITLE
Create an API to store NGO registration form

### PR DESCRIPTION
**Pull request by rtkanan**

### CODE REVIEW

File: controllers/NgoController.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  handleExport,
+  handleSave,
+  handleStatus,
+  handleStatusUpdate,
+} = require("../utils");
+
+// create a new ngo record
+exports.NgoStore = [
+  //auth,
+  body("mobile", "Mobile must not be empty.").isLength({ min: 1 }).trim(),
```
Comment (Kailash-Sankar) [inline controllers/NgoController.js:20]: we'll move this to a validation module, reuse it across the other forms

File: controllers/NgoController.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+// export data
+exports.export = [
+  auth,
+  function (req, res) {
+    handleExport(req, res, Ngo);
+  },
+];
+
+// TODO: not used for now
+exports.NgoStatus = [
```
Comment (Kailash-Sankar) [inline controllers/NgoController.js:44]: this routine is not applicable to ngo

File: models/NgoModel.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    act: { type: String, required: true },
+
+    name: { type: String, required: true }, // NGO's name
+    email: { type: String, required: true },
+    mobile: { type: Number, require: true },
+    alt_mob: { type: Number },
+
+    address: { type: String, required: false }, // NGO's address
+    reg: { type: String, required: true },
+    person: { type: String, required: true }, // NGO's point of contact (nodal person)
```
Comment (Kailash-Sankar) [inline models/NgoModel.js:17]: the form task had a pincode field

File: routes/main.js [hunk 4]
```diff
@@ -43,4 +44,16 @@ router.get("/request/status", RequestController.status);
...
+// save
+router.post("/ngo", NgoController.NgoStore);
+// search
+router.post("/ngo/search", NgoController.search);
+// export
+router.post("/ngo/export", NgoController.export);
+// overall status
+router.get("/ngo/status", NgoController.status);
+// Update the status of a appeal
+router.put("/ngo/update/:id", NgoController.updateStatus);
```
Comment (Kailash-Sankar) [inline routes/main.js:57]: would this route be used to verify/accept ngo registration

**Comment by rtkanan** [comment]
Closing this branch as the 'ngo' branch will be maintained separately to have all the NGO related enhancements.

**Pull request closed without merging**